### PR TITLE
(PE-13148) Add slingshot support to clj-puppetdb

### PR DESCRIPTION
**Pull request by kbrezina**
Originally, ExceptionInfo exceptions were thrown and caught.

### CODE REVIEW

**Comment by dmcpl** [comment]
Nice! LGTM! :+1:

**Comment by scottyw** [comment]
Sweet.

File: test/clj_puppetdb/http_test.clj [hunk 1]
```diff
@@ -56,53 +56,49 @@
...
-              (is (= (:msg info) response-data-encoded)))))))
+        (try+
+          (GET client q-path q-params)
+          (catch [] {:keys [status kind params endpoint host msg]}
+            (is (= status 400))
+            (is (= kind :puppetdb-query-error))
+            (is (= params q-params))
+            (is (= endpoint q-path))
+            (is (= host q-host))
+            (is (= msg response-data-encoded))))))
```
Comment (rlinehan) [inline test/clj_puppetdb/http_test.clj]: Another way we could do this if we wanted to would be

```clojure
  (is (thrown+? [:kind :puppetdb-query-error
                       :params q-params
                       :endpoint q-path
                       :host q-host
                       :status 400
                       :msg response-data-encoded]
                      (GET client q-path q-params)))
```

I'm not sure what I think about this, but just wanted to bring it up as an idea.

Comment (scottyw) [inline test/clj_puppetdb/http_test.clj]: I like that.

**Pull request merged**